src/nodes.coffee

```diff
@@ -3455,11 +3455,15 @@ exports.Assign = class Assign extends Base
   unfoldSoak: (o) ->
     unfoldSoak o, this, 'variable'
 
-  addScopeVariables: (o, {allowAssignmentToExpansion = no} = {}) ->
+  addScopeVariables: (o, {allowAssignmentToExpansion = no, allowAssignmentToNontrailingSplat = no, allowAssignmentToEmptyArray = no} = {}) ->
```

Comment:
I ran into another similar case: assignment to an empty array (eg `[] = b`) was failing during AST generation

So added `allowAssignmentToEmptyArray` here as well

I think I slightly prefer this explicit style of enumerating the assignability "exceptions", but if it feels cumbersome I guess we could just pass eg a `forAst: yes` flag instead and have `Arr::isAssignable()` be aware of what exactly is "assignable" during AST generation (ie when `forAst: yes`)?

Reply:
I think something like `allowAssignmentToEmptyArray` seems better because it’s clearer why behavior is branching. Maybe also add a comment as well explaining why this flag exists (i.e. your explanation above)?

Reply:
I agree, I think something like `allowAssignmentToEmptyArray` seems better because it’s clearer why behavior is branching. Maybe also add a comment as well explaining why this flag exists (i.e. your explanation above)?